### PR TITLE
Feat/sparql service category filter

### PR DESCRIPTION
**Pull request by fredericvilcot**
👉 This PR  aims to add a new filter regarding the `sparql` query onto the ontology database.

🧠 This new `byServiceCategoryFilter` allows to constraint the results for the share data - step1 (select a service storage) only to the objects in database which `serviceCategory` strictly equals `Storage` from the `vocabulary-category-service`.

🌶️ Technically, this PR adds 2 commands [`setByServiceCategoryFilter`- `resetByServiceCategoryFilter`] which return both `IO<void>` and add the desired filter onto the `sparql` gateway.

### CODE REVIEW

**Review by 0xLodz — APPROVED** [review]
LGTM! 🚀

File: src/infra/dataverse/sparql/sparqlGateway.ts [hunk 1]
```diff
@@ -39,6 +41,7 @@ export const sparqlGateway: DataversePort = {
       PREFIX datasetMetadata: <https://ontology.okp4.space/metadata/dataset/>
       PREFIX zoneMetadata: <https://ontology.okp4.space/metadata/zone/>
       PREFIX skos: <http://www.w3.org/2004/02/skos/core#>
+      PREFIX svccat: <https://ontology.okp4.space/thesaurus/service-category/>
```
Comment (ccamel) [inline src/infra/dataverse/sparql/sparqlGateway.ts:44]: To maintain consistency, I would recommend to standardize the style of prefixes within `RDF` ontologies. In your code, camel case style was used, while here lower case style is being utilized. Moreover, although prefixes can be named arbitrarily, it is generally a good thing to adopt standardized prefixes to ensure clarity and recognition (e.g. `foaf`, `rdfs`, `cc`).

I think we should rename the other prefixes to camel case for all prefixes.

**Review by lolottetheclash — APPROVED** [review]
Well done! 🚀

File: src/infra/dataverse/sparql/sparqlGateway.ts [hunk 2]
```diff
@@ -56,6 +59,7 @@ export const sparqlGateway: DataversePort = {
         }
         ${byType === 'all' ? '' : `FILTER ( ${byTypeFilter(byType)} )`}
         ${byProperty?.property ? `FILTER ( ${byPropertyFilter(byProperty)} )` : ''}
+        ${O.isSome(byServiceCategory) ? `FILTER ( ?topic = svccat:${byServiceCategoryFilter})` : ''}
```
Comment (ccamel) [inline src/infra/dataverse/sparql/sparqlGateway.ts:62]: It's indeed correct, but I would have seized the opportunity to chain options, with a `O.map` + a `O.getOrElse` that I fill is more functional. However, this decision ultimately comes down to personal preference. :wink:

**Review by ccamel — CHANGES_REQUESTED** [review]
That's fine, I just have a few minor remarks that I'll let you appreciate.

File: src/infra/dataverse/sparql/sparqlGateway.ts (hunk 1 above)
Comment (fredericvilcot) [reply src/infra/dataverse/sparql/sparqlGateway.ts:44]: Yes, of course 😌
I'll suggest to follow the `lowercase` convention used on the ontology classes.

File: src/domain/dataverse/domain.ts [hunk 3]
```diff
@@ -97,7 +98,7 @@ export const storeFactory = (
             limit: 20,
             hasNext: false,
             error: O.none,
-            filters: { byType: 'all', byProperty: null },
+            filters: { byType: 'all', byProperty: null, byServiceCategory: O.none },
```
[the remark this answers is not in the document]
Comment (fredericvilcot) [reply src/domain/dataverse/domain.ts:101]: Yep, you're right.
My first plan was to refactor code in a next PR, but we can put it here as there is no a huge impact on changes.

**Review by ccamel — APPROVED** [review]
Great! Thanks :+1:

**Pull request merged**